common/flatpak-run-sockets.c

```diff
@@ -186,6 +186,11 @@ flatpak_run_add_socket_args_environment (FlatpakBwrap         *bwrap,
       has_wayland = flatpak_run_add_wayland_args (bwrap, app_id, instance_id);
     }
 
+  if ((sockets & FLATPAK_CONTEXT_SOCKET_INHERIT_WAYLAND_SOCKET) == 0)
+    {
+      flatpak_bwrap_unset_env (bwrap, "WAYLAND_SOCKET");
```

Comment:
The app still inherits the socket this way, it just might not be aware which fd is referenced by the socket. We should also close the socket.

Reply:
Since we don't want to unset WAYLAND_SOCKET, I update to set FD_CLOEXEC on the fd instead of close it.

Otherwise, based on what you said following could happen:
1. thread 1 close fd.
2. thread 2 read wayland socket value .
3. some other thing cause a fd reuse.
4. thead 2 close a wrong fd.